IdentityServer/Startup.cs

```diff
@@ -77,36 +85,38 @@ public Startup(IConfiguration configuration, IWebHostEnvironment environment)
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddDbContext<IdentityDbContext>(o =>
+            {
+                o.UseSqlServer(Configuration.GetConnectionString("DefaultConnection"),
+                               sqlOptions => sqlOptions.EnableRetryOnFailure(50, TimeSpan.FromSeconds(30), null));
+            });
+
+            services.AddScoped<DbContext, IdentityDbContext>();
+            services.AddScoped<IIdentityUserService, IdentityUserService>();
+            services.AddScoped<IIdentityUserRepository, IdentityUserRepository>();
+
             // configures the OpenIdConnect handlers to persist the state parameter into the server-side IDistributedCache.
             services.AddOidcStateDataFormatterCache();
 
             services.AddControllersWithViews();
-            List<TestUser> testUsers = TestUsers.GetTestUsers(Environment.IsProduction());
 
             IIdentityServerBuilder builder = services.AddIdentityServer(options =>
-                                                     {
-                                                         options.Events.RaiseErrorEvents = true;
-                                                         options.Events.RaiseInformationEvents =
-                                                             true;
-                                                         options.Events.RaiseFailureEvents = true;
-                                                         options.Events.RaiseSuccessEvents = true;
-                                                         if(Environment.IsDevelopment())
-                                                         {
-                                                             options.IssuerUri = Config.Self.IssuerUri;
-                                                         }
-                                                         else
-                                                         {
-                                                             options.PublicOrigin = Config.Self.PublicOrigin;
-                                                         }
-                                                     })
-                                                     .AddTestUsers(testUsers);
-
-            // in-memory, code config
+            {
+                options.Events.RaiseErrorEvents = true;
+                options.Events.RaiseInformationEvents =
+                    true;
+                options.Events.RaiseFailureEvents = true;
+                options.Events.RaiseSuccessEvents = true;
+                if(Environment.IsDevelopment())
+                {
+                    options.IssuerUri = Config.Self.IssuerUri;
+                }
+            });
+
             builder.AddInMemoryIdentityResources(IdentityConfig.GetIdentityResources());
             builder.AddInMemoryApiResources(IdentityConfig.Apis);
             builder.AddInMemoryClients(IdentityConfig.Clients(Config));
```

Comment:
I'm unsure on why this is still in memory. I think for production the Identity Resources, Api Resources and the clients should also be persistent. Something along the lines of: 

```
            builder.AddOperationalStore(options =>
                options.ConfigureDbContext = builder =>
                    builder.UseSql(Configuration.GetConnectionString("DefaultConnection")));
            builder.AddConfigurationStore(options => options.ConfigureDbContext = dbContextOptionsBuilder =>     dbContextOptionsBuilder.UseSql(Configuration.GetConnectionString("DefaultConnection")));
            builder.AddAspNetIdentity<ApplicationUser>();
            builder.AddProfileService<IdentityProfileService>();
```

This is also what IdentityServer recommends for production scenarios. 
I don't think we really need in-memory for local development either seeing as we want developers to work  as close as possible to production level.

Reply:
the reason why this was kept in memory is that there is currently no UI where you can modify these values and running queries on a production database is not something i would encourage.

Reply:
I'll make a new issue for next semester to add a proper Admin UI to manage clients, Scopes and Resources properly.